src/libstd/panicking.rs

```diff
@@ -41,12 +65,10 @@ use realstd::io::set_panic;
 // hook up these functions, but it is not this day!
 #[allow(improper_ctypes)]
 extern "C" {
-    fn __rust_maybe_catch_panic(
-        f: fn(*mut u8),
-        data: *mut u8,
-        data_ptr: *mut usize,
-        vtable_ptr: *mut usize,
-    ) -> u32;
+    /// The payload ptr here is actually the same as the payload ptr for the try
+    /// intrinsic (i.e., is really `*mut [u64; 2]` or `*mut *mut u8`).
+    #[unwind(allowed)]
```

Comment:
This is not strictly true today (e.g., `emcc` uses Option::unwrap, as does the gcc implementation).

Reply:
The gcc implementation no longer uses unwrap (see the commit I just pushed). And in any case, any panic in this function would cause a double panic and abort, and will never unwind.